README.md

```diff
@@ -451,11 +451,11 @@ stringData:
 ```
 ###### Custom Templates
 
-For additional flexibility, you can model your `Secret` resources according to your specific needs.
+For additional flexibility, you can model your `Secret` resources according to your specific needs.<br>
 To generate a custom-formatted `Secret`, use the `secretTemplate` attribute in the `ServiceBinding` spec.
-The value of the `secretTemplate` attribute must be a Go template, The generated output of the template should be in YAML format, representing the desired 'Secret'.
-The secret will be created with the data provided in the template and if `secretKey` and `secretRootKey` attributes are provided, they will be ignored.
-To include the default data in the secret (as provided by us), you should not include the `data` and `stringData` fields in your `secretTemplate`.
+The value of the `secretTemplate` attribute must be a Go template, The generated output of the template should be in YAML format, representing the desired 'Secret', 
```

Comment:
@I065450  proposal:

The value of the 'secretTemplate' attribute must be a Go template. It is mandatory to specify in the output format section of the template YAML as a format for the Secret.

`with . format: YAML`  - not sure how exactly it looks but something along these lines.

Note that you should also specify which attributes of your customized Secret resource should be base64 encoded, and which should be a plain string. 
For example:
s. apiVersion: v1
kind: Secret
metadata:
  name: my-secret
  namespace: default
data:
  username: {{ .Username | b64encode }} 
  password: {{ .Password | b64encode }} 
stringData:
  api_url: {{ .APIURL }}

If you do not specify this, the template will use default settings as defined by us for the attributes (from what I see, it is only stringData?)

Reply:
*will use the encoding settings as provided by us.

Reply:
the value of .... should be a YAML file of the Secret structure specified 
labels annotations - see the example 
The template should be in the YAML format with the structure of the Kubernetes Secret. Note that in the Secret Metadata, you can customize only its labels and annotations.

Reply:
if a user adds secretKey or secretRootKey or nothing but specifies stringdata or data in the template, then the template takes precedence 
if they don't use data or stringdata in the template, then secretKey/secretRootKey and none are applied cause they only customized the metadata part of the Secret structure. 
format of stringdata will be one of the three (key-value pairs, credentials as JSON, credentials and service info as JSON)

Reply:
provide explanation - bulleted list